test/avi/test_helpers.clj

```diff
@@ -140,6 +140,25 @@
   [expected]
   (line nil expected))
 
+(defn terminal-buffer
```

Comment:
Hey... sorry about the delay.  I was busy, then it got lost in the inbox, and I'm just catching up.

Also, I'm trying out this new Github review thing.  Let's see if it works.  I hope it will let me accept the pull request without losing the feedback.

I'd try to remove the duplicated code between `line` and `terminal-buffer`.  Perhaps `line`, `message-line`, `terminal`, `terminal-buffer` would all operate in terms of a more general `lines` that has all the code, and can take `:buffer` as a keyword as well as line numbers, `nil`, and `:message`.